### PR TITLE
Add All Hands Retrospective

### PR DESCRIPTION
**Pull request by XAMPPRocky**
### [Rendered](https://github.com/XAMPPRocky/blog.rust-lang.org/blob/all-hands/posts/inside-rust/2020-03-19-all-hands-retrospective.md)

r? @nikomatsakis 

cc @rust-lang/core

### CODE REVIEW

File: posts/inside-rust/2020-03-19-all-hands-retrospective.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+quite fit into the budgets and categories it would have to come from. We also
+had some uncertainty about how much "participation" to offer in exchange for
+sponsorship, since we didn't want to give the impression that one could just
+"buy your way" into the All Hands — at the same time, there is clear value to
+the Rust organisation in having sponsors participate.
+
+## The Future
+
+Right now, we're currently organising reimbursing anyone who had purchased non
+refundable ticket. We're also going to take the time to step back an re
```
Comment (bjorn3) [inline posts/inside-rust/2020-03-19-all-hands-retrospective.md]:
```suggestion
refundable ticket. We're also going to take the time to step back and re-
```

**Comment by lzutao** [comment]
The rendered links is broken.

**Comment by bjorn3** [comment]
The rendered link doesn't work.

**Comment by XAMPPRocky** [comment]
Thanks, forgot to update it with the file name change.

File: posts/inside-rust/2020-03-19-all-hands-retrospective.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+opted for the painful but ultimately safer route: cancelling the event.
+
+**Addendum** Since we cancelled the All Hands, outbreak of the [COVID-19] in
+Europe in the recent weeks, would have caused the event to be cancelled
+regardless. In a weird twist of fate, we were fortunate to cancel when we did.
+
+[covid-19]: https://en.wikipedia.org/wiki/2019–20_coronavirus_pandemic
+
+## Lessons Learned
+**Bound uncertainty and and total cost.** — Promising to cover all travel with
```
Comment (CryZe) [inline posts/inside-rust/2020-03-19-all-hands-retrospective.md]:
```suggestion
**Bound uncertainty and total cost.** — Promising to cover all travel with
```

File: posts/inside-rust/2020-03-19-all-hands-retrospective.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+did not have access to any expense reports from the previous years. We had a
+survey we could use to estimate the number of attendees and a rough idea where
+they were coming from. We knew the costs of the venue. We added fudge factors.
+We thought we'd done a good job, though it would turn out that we were quite
+off in our calculations.
+
+Based on our estimated budget, it was clear that Mozilla was not going to be
+able to cover the costs alone. It seemed like a good opportunity to try and
+build relationships with some of the companies that have started to use Rust
+by approaching them for sponsorships. We weren't sure, though, how man
```
Comment (blankname) [inline posts/inside-rust/2020-03-19-all-hands-retrospective.md]:
```suggestion
by approaching them for sponsorships. We weren't sure, though, how many
```

File: posts/inside-rust/2020-03-19-all-hands-retrospective.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+**Location, location, location** — It was hard to find a good location. While
+Thessaloníki was cheap and great in some ways, we encountered some logistical
+hiccups that made it less than ideal. For example; we found that the cost of a
+lot of the flights for our international members were quite high compared to
+other European
+cities.
+
+**Know your audience** — Some companies weren't interested in sponsoring
+because the event wasn't sufficiently high profile, or because the "ROI" didn't
```
Comment (pietroalbini) [inline posts/inside-rust/2020-03-19-all-hands-retrospective.md]: Maybe let's note what ROI means for the people unfamiliar with the abbreviation.

File: posts/inside-rust/2020-03-19-all-hands-retrospective.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+("Return On Investment") didn't quite fit into the budgets and categories it
+would have to come from. We also had some uncertainty about how much
+"participation" to offer in exchange for sponsorship, since we didn't want to
+give the impression that one could just "buy your way" into the All Hands — at
+the same time, there is clear value to the Rust organisation in having
+sponsors participate.
+
+## The Future
+
+Right now, we're currently organising reimbursing anyone who had purchased non
```
Comment (nikomatsakis) [inline posts/inside-rust/2020-03-19-all-hands-retrospective.md]: I think we should add a sentence crediting Mozilla for paying the reimbursements here (or somewhere).

File: posts/inside-rust/2020-03-19-all-hands-retrospective.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+information to repay people who had booked travel.
+
+In retrospect, it's clear that this was premature. It seemed like all the
+pieces were in place, and we knew that if people didn't start booking tickets,
+they never would. But the pieces were not in place, not actually, and we clearly
+should have taken this moment ask whether we were really going to be able to
+pull this off.
+
+### Realising it won't work.
+Finally, this all started to unravel in a about a month away from the event.
```
Comment (nrc) [inline posts/inside-rust/2020-03-19-all-hands-retrospective.md]:
```suggestion
This all started to unravel about a month before the event.
```

File: posts/inside-rust/2020-03-19-all-hands-retrospective.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+* If we continue with the event, there was the chance that the logistics would
+  just fail to come together. That would be a true disaster.
+* Further, we might not be able to raise enough money to be able to pay for
+  the entirety of people's travel and lodging as we had promised. This would be
+  unfortunate, though perhaps something people could live with.
+* On the other hand, if we cancel, we have to deal with the fact that people
+  had already started booking travel.
+
+After much analysis and discussion with the Rust core team, we ultimately
```
Comment (nrc) [inline posts/inside-rust/2020-03-19-all-hands-retrospective.md]: If you can remember, it would be nice to add the reasoning here.

File: posts/inside-rust/2020-03-19-all-hands-retrospective.md [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+**Addendum** Since we cancelled the All Hands, outbreak of the [COVID-19] in
+Europe in the recent weeks, would have caused the event to be cancelled
+regardless. In a weird twist of fate, we were fortunate to cancel when we did.
+
+[covid-19]: https://en.wikipedia.org/wiki/2019–20_coronavirus_pandemic
+
+## Lessons Learned
+**Bound uncertainty and total cost.** — Promising to cover all travel with
+no kind of upper limit was unwise. It had worked in the past, perhaps, but the
+event was growing, and that was partly luck.
```
Comment (nrc) [inline posts/inside-rust/2020-03-19-all-hands-retrospective.md]: I think it was largely due to a different funding model rather than luck

File: posts/inside-rust/2020-03-19-all-hands-retrospective.md [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+other European
+cities.
+
+**Know your audience** — Some companies weren't interested in sponsoring
+because the event wasn't sufficiently high profile, or because the "ROI"
+("Return On Investment") didn't quite fit into the budgets and categories it
+would have to come from. We also had some uncertainty about how much
+"participation" to offer in exchange for sponsorship, since we didn't want to
+give the impression that one could just "buy your way" into the All Hands — at
+the same time, there is clear value to the Rust organisation in having
```
Comment (nrc) [inline posts/inside-rust/2020-03-19-all-hands-retrospective.md]: I would rephrase along to make clear that the people we approached to sponsor are people we would want at the event in any case.

File: posts/inside-rust/2020-03-19-all-hands-retrospective.md [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+sponsors participate.
+
```
Comment (nrc) [inline posts/inside-rust/2020-03-19-all-hands-retrospective.md]: I would add that organising such an event is more complex, and takes longer than we thought. Relatively minor issues like having a bank account blew up into big things and we didn't have the time or resources to deal with them

File: posts/inside-rust/2020-03-19-all-hands-retrospective.md [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+In retrospect, it's clear that this was premature. It seemed like all the
+pieces were in place, and we knew that if people didn't start booking tickets,
+they never would. But the pieces were not in place, not actually, and we clearly
+should have taken this moment ask whether we were really going to be able to
+pull this off.
+
+### Realising it won't work.
+Finally, this all started to unravel in a about a month away from the event.
+In short, we encountered some logistical hurdles, which we won't go into, but
```
Comment (nrc) [inline posts/inside-rust/2020-03-19-all-hands-retrospective.md]: I would expand a little - something like: business and legal issues which were insurmountable due to other organizations' constraints.

File: posts/inside-rust/2020-03-19-all-hands-retrospective.md [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+which also prompted us to review the finances in more detail. This revealed
+that our estimates were off. Thus, we were faced with a tough decision:
+
+* If we continue with the event, there was the chance that the logistics would
+  just fail to come together. That would be a true disaster.
```
Comment (nrc) [inline posts/inside-rust/2020-03-19-all-hands-retrospective.md]:
```suggestion
  just fail to come together. That would leave attendees in Greece, out of pocket, and without an All Hands to attend.
```

File: posts/inside-rust/2020-03-19-all-hands-retrospective.md [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+* Further, we might not be able to raise enough money to be able to pay for
+  the entirety of people's travel and lodging as we had promised. This would be
+  unfortunate, though perhaps something people could live with.
+* On the other hand, if we cancel, we have to deal with the fact that people
+  had already started booking travel.
```
Comment (nrc) [inline posts/inside-rust/2020-03-19-all-hands-retrospective.md]: add something like "and knowing that it *might* have worked if only we had tried" or similar?

File: posts/inside-rust/2020-03-19-all-hands-retrospective.md (hunk 8 above)
Comment (XAMPPRocky) [reply posts/inside-rust/2020-03-19-all-hands-retrospective.md]: I think luck here is referring to other factors than just funding. For example, some of the costs to pay for people's travel was definitely offset by hosting in Berlin which was/is already pretty populated with Rust org members.

**Comment by XAMPPRocky** [comment]
Okay, I think I've addressed all feedback, and this should be ready re-review/merge.
